src/librustc_typeck/check/coercion.rs

```diff
@@ -555,9 +555,19 @@ impl<'f, 'gcx, 'tcx> Coerce<'f, 'gcx, 'tcx> {
                 ty::Predicate::Trait(ref tr) if traits.contains(&tr.def_id()) => {
```

Comment:
Note to self: double check how these traits work, do we always want to apply this test here?